main.rb

```diff
@@ -0,0 +1,125 @@
+# rubocop: disable Metrics/CyclomaticComplexity, Metrics/MethodLength,Metrics/PerceivedComplexity
+
+module Enumerable
+  def my_each
+    return enum_for(:my_each) unless block_given?
+
+    0.upto(to_a.size - 1) do |n|
+      yield to_a[n]
+    end
+    to_a
+  end
+
+  def my_each_with_index
+    return enum_for(:my_each) unless block_given?
+
+    0.upto(to_a.size - 1) do |i|
+      yield to_a[i], i
+    end
+    to_a
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    result = []
+    my_each { |i| result << i if yield(i) }
+    result
+  end
+
+  def my_all?
+    return true unless block_given?
+
+    result = 0
+    while result < size
+      return false unless yield(self[result])
+
+      result += 1
+    end
+
+    true
+  end
```

Comment:
enumerable my_all? returns true when none of the collection members are false or nil.
```
=> arr = [0, false, 7, 4, 5, 8, 9, 1, 9, 0]
=> arr.all?
=> false
=> arr.my_all?
=> true
```

Expected to be false.